app/helpers/queries_helper.rb

```diff
@@ -76,4 +76,24 @@ def retrieve_query
 
     @query
   end
+
+  def visible_queries
+    unless @visible_queries
+      # User can see public queries and his own queries
+      visible = ARCondition.new(['is_public = ? OR user_id = ?', true, (User.current.logged? ? User.current.id : 0)])
+      # Project specific queries and global queries
+      visible << (@project.nil? ? ['project_id IS NULL'] : ['project_id IS NULL OR project_id = ?', @project.id])
+      @visible_queries = Query.where(visible.conditions)
+                              .order('name ASC')
+                              .select(:id, :name, :is_public, :project_id)
+    end
+    @visible_queries
+  end
+
+  def grouped_query_options
```

Comment:
@ulferts it appears that `grouped_query_options` is no longer being used anywhere.